specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/apimsettings.json

```diff
@@ -0,0 +1,144 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ApiManagementClient",
+    "description": "Use these REST APIs for performing operations on settings entity associated with your Azure API Management deployment. Using this entity you can manage properties and configuration that apply to the entire API Management service instance.",
+    "version": "2020-06-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/settings": {
+      "get": {
+        "tags": [
+          "TenantSettings"
+        ],
+        "operationId": "TenantSettings_ListByService",
+        "description": "Returns empty list.",
+        "x-ms-examples": {
+          "ApiManagementListTenantSettings": {
+            "$ref": "./examples/ApiManagementListTenantSettings.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "Not used"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Lists a collection of TenantSettings entities.",
+            "schema": {
+              "$ref": "./definitions.json#/definitions/TenantSettingsCollection"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-odata": "./definitions.json#/definitions/TenantSettingsContract"
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/settings/{settingsType}": {
```

Comment:
>settingsType [](start = 138, length = 12)

I would expected under settings there would be an array of settingsType. From the model provided in  swagger and examples it's not clear. Please share more details.

Reply:
Same issue - we don't have collection of settings. We have one entity describing all the settings of the tenant. Customers don't use this endpoint but we use it in portal so it must be ARM compliant. What we do - introduced EMPTY collection "/settings"   and one particular endpoint "/settings/public" which returns the desired entity.
Same question: Please advise what is the proper way to implement GET endpoint for the entity without collection.